include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_pressure_correction.cpp

```diff
@@ -353,10 +353,9 @@ TimeIntBDFPressureCorrection<dim, Number>::momentum_step()
   }
 
   /*
-   *  if a variable viscosity is used: update
-   *  viscosity model before calculating rhs_momentum
+   *  explicit variable viscosity update executed prior to calculation of rhs_momentum
    */
-  if(this->param.viscosity_is_variable())
+  if(this->param.viscosity_is_variable() and not this->param.viscous_term_is_nonlinear())
```

Comment:
same as here https://github.com/exadg/exadg/pull/479/files#r1293216935

Reply:
also corrected the condition in the `driver_steady_problems.cpp`